edgexfoundry/edgex-compose
# refactor: Rework for standard MESSAGEBUS config for all services

This PR is dependent on the following PRs
https://github.com/edgexfoundry/edgex-go/pull/4268
https://github.com/edgexfoundry/device-sdk-go/pull/1273
https://github.com/edgexfoundry/app-functions-sdk-go/pull/1255
https://github.com/edgexfoundry/app-service-configurable/pull/499
https://github.com/edgexfoundry/device-rest-go/pull/236
https://github.com/edgexfoundry/device-virtual-go/pull/327
https://github.com/edgexfoundry/device-modbus-go/pull/399
https://github.com/edgexfoundry/device-onvif-camera/pull/210

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  TBD


## Testing Instructions
from builder run `make run no-secty ds-rest ds-virtual ds-modbus ds-onvif-camera`
Verify all services bootstrap properly and connect to the MessageBus and have proper topics for publish and subscribe.
- example log for topics: 
```
level=INFO ts=2023-01-11T15:56:22.2096166Z app=app-sample source=messaging.go:104 msg="Connected to redis Message Bus @ redis://localhost:
6379 using topics [map[PublishTopic:sample SubscribeTopics:edgex/events/#]] with AuthMode='usernamepassword'"
```
